Sources/SwifterSwift/UIKit/UILabelExtensions.swift

```diff
@@ -9,7 +9,55 @@
 #if canImport(UIKit) && !os(watchOS)
 import UIKit
 
+// MARK: - Properties
+public extension UILabel {
+
+    /// SwifterSwift: Required height for a label
+    var requiredHeight: CGFloat {
+        let label = UILabel(frame: CGRect(x: 0, y: 0, width: frame.width, height: CGFloat.greatestFiniteMagnitude))
+        label.numberOfLines = 0
+        label.lineBreakMode = NSLineBreakMode.byWordWrapping
+        label.font = font
+        label.text = text
+        label.attributedText = attributedText
+        label.sizeToFit()
+        return label.frame.height
+    }
+}
+
 // MARK: - Methods
+public extension UILabel {
+
+    /// SwifterSwift:  Calculate and return the size that best fits an attributed string, given the specified constraints on size
+    ///
+    ///   let yourAttributedString = ...
+    ///   let yourConstraintSize = ...
+    ///   let size = UILabel.size(thatFitsAttributedString: yourAttributedString, withConstraints: yourConstraintSize)
+    ///
+    /// - Parameters:
+    ///   - attributedString: The attributed string.
+    ///   - size: The maximum dimensions used to calculate size.
+    /// - Returns: The size that fits the attributed string within the specified constraints.
+    static func size(thatFitsAttributedString attributedString: NSAttributedString,
+                     withConstraints size: CGSize) -> CGSize {
+
+        guard attributedString.length > 0 else { return CGSize.zero}
+
+        let framesetter = CTFramesetterCreateWithAttributedString(attributedString)
+
+        let calculatedSize = CTFramesetterSuggestFrameSizeWithConstraints(framesetter,
```

Comment:
Since this is coming from CoreText, please wrap the function with a `canImport` check

```swift
public extension UILabel {
    #if canImport(CoreText)
    func ...
    #endif
}
```